internal/controllers/device_data_controller.go

```diff
@@ -398,15 +400,62 @@ func (udc *UserDevicesController) GetUserDeviceErrorCodeQueries(c *fiber.Ctx) er
 			return err
 		}
 
-		queries = append(queries, GetUserDeviceErrorCodeQueriesResponseItem{
+		userDeviceresp := GetUserDeviceErrorCodeQueriesResponseItem{
 			ErrorCodes:  ercJSON,
 			RequestedAt: erc.CreatedAt,
```

Comment:
We talked about populating `ClearedAt` here. Can we do that? `null.Time` has a [`Ptr`](https://pkg.go.dev/github.com/volatiletech/null/v8#Time.Ptr) method that should make this simple.